apps/dsky_clock/app.js

```diff
@@ -0,0 +1,373 @@
+require("Font7x11Numeric7Seg").add(Graphics);
+require("FontTeletext5x9Ascii").add(Graphics);
+
+const Light_on='#fff';
+const Light_off='#554';
+const Light_warn='#f90';
+const Light_COMPACTY='#0F0';
+const Light_width=43;
+const Light_height=25;
+const EL7_height=30;
+const LightFont='Teletext5x9Ascii';
+const DataFont='7x11Numeric7Seg:2';
+var mode = 0;
+
+var layout = new Layout(
+  {type:"h", c:[
+    {type:"",width:6},
+    { type:"v", c: [
+        {type:"txt", font:LightFont, col:"#000", bgCol:"#555", id:'L1', label:"UPLINK\nACTY", width:Light_width, height:Light_height},
+        {type:"txt", font:LightFont, col:"#000", bgCol:"#555", id:'L2', label:"TEMP", width:Light_width, height:Light_height },
+        {type:"txt", font:LightFont, col:"#000", bgCol:"#555", id:'L3', label:"GIMBAL\nLOCK", width:Light_width, height:Light_height },
+        {type:"txt", font:LightFont, col:"#000", bgCol:"#555", id:'L4', label:"STBY", width:Light_width, height:Light_height },
+        {type:"txt", font:LightFont, col:"#000", bgCol:"#555", id:'L5', label:"PROG", width:Light_width, height:Light_height },
+        {type:"txt", font:LightFont, col:"#000", bgCol:"#eee", id:'L6', label:"OPR ERR", width:Light_width, height:Light_height },
+    ]},
+    { type:"", width:20},
+    { type:"v", c: [
+      {type:"",height:2},
+      {type:"h", c: [
+        {type:"", width:50},{type:"txt", font:"6x8", col:"#000",bgCol:"#0F0", label:"PROG", width:25, height:10},
+      ]},
+      {type:"h", c: [
+        {type:"",width:10},
+        {type:"txt", font:"6x8", col:"#000", bgCol:"#000", id:"COMPACTY", label:"COMP\nACTY", width:26, height:26 },
+        {type:"",width:17},
+        {type:"txt", font:DataFont, col:"#0F0", bgCol:"#000",label:"00", id:"PROG", fillx:1, height:EL7_height },
+      ]},
+      {type:"",height:1},
+      {type:"h", c: [
+        {type:"txt", font:"6x8", col:"#000", bgCol:"#0F0",label:"VERB", width:25, height:10},
+        {type:"",width:30},
+        {type:"txt", font:"6x8", col:"#000",bgCol:"#0F0",label:"NOUN", width:25, height:10},
+      ]},
+      {type:"h", c: [
+          {type:"txt", font:DataFont, col:"#0F0", bgCol:"#000", label:"00", id:"VERB", fillx:1, height:EL7_height},
+          {type:"txt", font:DataFont, col:"#0F0", bgCol:"#000", label:"00", id:"NOUN", fillx:1, height:EL7_height}, 
+      ]},
+      { type:"",bgCol:'#070', width:80, height:2 },
+        {type:"txt", font:DataFont, col:"#0F0", bgCol:"#000", label:"00000", id:"R1", halign:1, fillx:1, height:EL7_height},
+        {type:"txt", font:DataFont, col:"#0F0", bgCol:"#000", label:"00000", id:"R2", halign:1, fillx:1, height:EL7_height},
+        {type:"txt", font:DataFont, col:"#0F0", bgCol:"#000", label:"00000", id:"R3", halign:1, fillx:1, height:EL7_height},
+      ]},
+    {type:"",width:5},
+  ],
+  lazy:true});
+layout.update();
+
+//support functions
+
+function getWeather() {
+  var weather = require("Storage").readJSON('weather.json', 1).weather;
```

Comment:
I don't have the weather app installed so I trigger this: 

```terminal
Uncaught Error: Can't read property 'weather' of undefined
 at line 70 col 63 in dsky_clock.app.js
...readJSON('weather.json', 1).weather;
                              ^
in function "getWeather" called from line 281 col 26 in dsky_clock.app.js
let weather=getWeather();
                       ^
in function "mode_weather" called from line 247 col 20 in dsky_clock.app.js
      mode_weather();
                   ^
in function "drawAlt" called from line 361 col 15 in dsky_clock.app.js
  drawAlt(mode);
              ^
in function called from system
> 
```

So some way of handling this situation is needed.